src/TicTacToe.Console/TicTacToe.Console/Players/PlayerRegistrationService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TicTacToe.Foundation.Interfaces;
+using TicTacToe.Foundation.Figures;
+using TicTacToe.Console.Interfaces;
+
+namespace TicTacToe.Console.Players
+{
+    public class PlayerRegistrationService : IPlayerRegistrationService
+    {
+        private readonly IPlayerFactory _playerFactory;
+        private readonly IConsole _console;
+
+
+        public PlayerRegistrationService(
+            IPlayerFactory playerFactory,
+            IConsole console)
+        { 
+            _playerFactory = playerFactory;
+            _console = console;
+        }
+
+
+        public IPlayer RegisterPlayer(IReadOnlyCollection<FigureType> availableFigureTypes)
+        {
+            _console.WriteLine("Please, enter player's name: ");
+            var name = _console.ReadLine();
+            var figureType = ChooseFigure(availableFigureTypes);
+
+            return _playerFactory.CreatePlayer(name, figureType);
+        }
+
+
+        private FigureType ChooseFigure(IReadOnlyCollection<FigureType> availableFigureTypes)
+        {
+            if (!availableFigureTypes.Any())
+            {
+                throw new InvalidOperationException("There is no more available figure types");
+            }
+            if (availableFigureTypes.Count == 1)
+            {
+                _console.WriteLine($"There is only one figure type left. You figure type: {availableFigureTypes.First().ToString()}");
+
+                return availableFigureTypes.Single();
+            }
+
+            _console.WriteLine("Available figures types:");
+            
+            var i = 1;
+            availableFigureTypes.ToList().ForEach(ft => _console.WriteLine($"{i++}) {ft}"));
```

Comment:
1) Please, move `.ForEach` to common assembly, so we can iterate any enumerable.
2) don't declare outside `i` variable